src/main/java/org/trips/service_framework/heplers/CmsHelper.java

```diff
@@ -0,0 +1,87 @@
+package org.trips.service_framework.heplers;
+
+import org.springframework.stereotype.Service;
+import org.trips.service_framework.constants.CmsConstants;
+import org.trips.service_framework.dtos.SkuAttributes;
+import org.apache.commons.text.StringSubstitutor;
+import org.trips.service_framework.utils.CmsUtils;
+
+import java.util.*;
+
+@Service
+public class CmsHelper {
+
+    public SkuAttributes getAttributesFromMap(Map<String, String> attributeMap) {
+        return SkuAttributes.builder()
+                .species(attributeMap.get("species"))
+                .productType(attributeMap.get("product_type"))
+                .spec(attributeMap.get("spec"))
+                .catchType(attributeMap.get("catch_type"))
+                .freezingMethod(attributeMap.get("freezing_method"))
+                .packing(attributeMap.get("packing"))
+                .glazingPercentage(attributeMap.get("glazing_percentage"))
+                .unitWeight(attributeMap.get("unit_weight"))
+                .quantityPerUnit(attributeMap.get("quantity_per_unit"))
+                .unitPerCarton(attributeMap.get("unit_per_carton"))
+                .treatment(attributeMap.get("treatment"))
+                .grade(attributeMap.get("grade"))
+                .quality(attributeMap.get("quality"))
+                .build();
+    }
+
+    public Map<String, Object> createRequestMap(String name, String value, String operator, boolean isAttribute) {
+        Map<String, Object> requestMap = new HashMap<>();
+        requestMap.put("name", name);
+        requestMap.put("value", CmsUtils.sanitizeInput(name, value));
+        requestMap.put("operator", operator);
+        requestMap.put("isAttribute", isAttribute);
+
+        return requestMap;
+    }
+
+    public Map<String, Object> getSearchQueryFromSkuCode(String code) {
+        Map<String, Object> requestMap = createRequestMap("code", code, "EQ", false);
+        return Map.of("searchQuery", Map.of("filters", List.of(requestMap)));
+    }
+
+    public Map<String, Object> getSearchQueryFromAttributes(SkuAttributes attributes) {
+        List<Map<String, Object>> attributeList = new ArrayList<>();
+        attributeList.add(createRequestMap("species", attributes.getSpecies(), "EQ", true));
```

Comment:
Since `operator` and `isAttribute` are always the same (and will remain that way for the foreseeable future), can we create a list with the specific fields we need from `SkuAttributes` and then write a function that loops over that list and transforms it into `attributeList`?

Something like
```
List<Map<String, String>> rawAttributeList = List.of(attribute.getSpecies(), attribute.getSpec()...);
List<Map<String, Object>> processedAttributeList = transformSkuAttributes(rawAttributeList);
 
private List<Map<String, Object>> transformSkuAttributes(List<Map<String, String>> rawAttributeList) {
    for (Map attribute: rawAttributeList) {
        // use createRequestMap here
    }
}
```